client/components/app.jsx

```diff
@@ -52,14 +55,21 @@ export default class App extends React.Component {
     if (this.state.view.name === 'catalog') {
       return (
         <div>
-          <Header cartItemCount={this.state.cart.length} />
+          <Header setView={this.setView} cartItemCount={this.state.cart.length} />
           <ProductList setView={this.setView} />
         </div>
       );
+    } else if (this.state.view.name === 'cart') {
+      return (
+        <div>
+          <Header setView={this.setView} cartItemCount={this.state.cart.length} />
+          <CartSummary setView={this.setView} cart={this.state.cart} />
+        </div>
+      );
     } else {
       return (
         <div>
-          <Header cartItemCount={this.state.cart.length} />
+          <Header setView={this.setView} cartItemCount={this.state.cart.length} />
           <ProductDetails addToCart={this.addToCart} params={this.state.view.params} setView={this.setView} />
         </div>
       );
```

Comment:
The purpose of the `setView` method is only to call `setState` and change the view property in the state object.

Your logic on deciding what to render depending on the `view` property of the `state` object should be ran in the `render` method.